docker/docker_image_dest.go

```diff
@@ -338,6 +346,21 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
 			logrus.Debugf("Error parsing BlobInfoCache location reference: %s", err)
 			continue
 		}
+		compressionOperation, compressionAlgorithm, err := blobinfocache.OperationAndAlgorithmForCompressor(candidate.CompressorName)
+		if err != nil {
+			logrus.Debugf("OperationAndAlgorithmForCompressor Failed: %v", err)
+			continue
+		}
+		opts := options
+		opts.OriginalCompression = compressionAlgorithm
```

Comment:
The value for this candidate is not “original blob’s compression”.

Split the helper into two instead?

```go
func BlobMatchesRequiredCompression(opts, alghorithmOrNil) {
   …
}

func OriginalBlobMatchesRequiredCompression(opts) {
   return BlobMatchesRequiredCompression(opts, opts.OriginalAlgorithm)
}
```
Then most transports can use `Original…`, and this path the other one.